package NKikimrMemory;
option java_package = "ru.yandex.kikimr.proto";

message TMemoryStats {
    optional uint64 AnonRss = 1;
    optional uint64 CGroupLimit = 2;
    optional uint64 MemTotal = 3;
    optional uint64 MemAvailable = 4;

    optional uint64 AllocatedMemory = 5;
    optional uint64 AllocatorCachesMemory = 6;

    optional uint64 HardLimit = 7;
    optional uint64 SoftLimit = 8;
    optional uint64 TargetUtilization = 9;

    reserved 10;
    reserved 11;
    reserved 12;
    optional uint64 ExternalConsumption = 13;

    optional uint64 SharedCacheConsumption = 14;
    optional uint64 SharedCacheLimit = 15;

    optional uint64 MemTableConsumption = 16;
    optional uint64 MemTableLimit = 17;

    optional uint64 QueryExecutionConsumption = 18;
    optional uint64 QueryExecutionLimit = 19;
}
